src/gsi/general_read_gfsatm.f90

```diff
@@ -3588,99 +3588,99 @@ subroutine general_read_gfsatm_allhydro_nc(grd,sp_a,filename,uvflag,vordivflag,z
-         ! rain number concentration.
-         if ( diff_res ) then
-            grid_b=rwork3d0(:,:,1)
-            vector(1)=.false.
-            call fill2_ns(grid_b,grid_c(:,:,1),latb+2,lonb)
-            call g_egrid2agrid(p_high,grid_c,grid2,1,1,vector)
-            do kk=1,grd%itotsub
-               i=grd%ltosi_s(kk)
-               j=grd%ltosj_s(kk)
-               work(kk)=grid2(i,j,1)
-            enddo
-         else
-            grid=rwork3d0(:,:,1)
-            call general_fill_ns(grd,grid,work)
-         endif
-      endif
-      if ( icount == icm .or. k==nlevs) then
-         call general_reload2(grd,g_z,g_ps,g_tv,g_vor,g_div,g_u,g_v,g_q,g_oz, &
-              g_ql,g_qi,g_qr,g_qs,g_qg,icount,iflag,ilev,work,uvflag,vordivflag,g_ni,g_nr)
-          endif
-       enddo ! do k=1,nlevs
+          do k=1,nlevs
+             icount=icount+1
+             iflag(icount)=16
+             ilev(icount)=k
+             kr = levs+1-k ! netcdf is top to bottom, need to flip
 
-!   do k=1,nlevs
-!      icount=icount+1
-!      iflag(icount)=17
-!      ilev(icount)=k
-!      kr = levs+1-k ! netcdf is top to bottom, need to flip
-!
-!      if (mype==mype_use(icount)) then
-!         call read_vardata(filges, 'cld_amt', rwork3d0, nslice=kr, slicedim=3)
-!         ! Cloud amount (cloud fraction). 
-!         if ( diff_res ) then
-!            grid_b=rwork3d0(:,:,1)
-!            vector(1)=.false.
-!            call fill2_ns(grid_b,grid_c(:,:,1),latb+2,lonb)
-!            call g_egrid2agrid(p_high,grid_c,grid2,1,1,vector)
-!            do kk=1,grd%itotsub
-!               i=grd%ltosi_s(kk)
-!               j=grd%ltosj_s(kk)
-!               work(kk)=grid2(i,j,1)
-!            enddo
-!         else
-!            grid=rwork3d0(:,:,1)
-!            call general_fill_ns(grd,grid,work)
-!         endif
+             if (mype==mype_use(icount)) then
+                call read_vardata(filges, 'nconrd', rwork3d0, nslice=kr, slicedim=3)
+                ! rain number concentration.
+                if ( diff_res ) then
+                   grid_b=rwork3d0(:,:,1)
+                   vector(1)=.false.
+                   call fill2_ns(grid_b,grid_c(:,:,1),latb+2,lonb)
+                   call g_egrid2agrid(p_high,grid_c,grid2,1,1,vector)
+                   do kk=1,grd%itotsub
+                      i=grd%ltosi_s(kk)
+                      j=grd%ltosj_s(kk)
+                      work(kk)=grid2(i,j,1)
+                   enddo
+                else
+                   grid=rwork3d0(:,:,1)
+                   call general_fill_ns(grd,grid,work)
+                endif
+             endif
+             if ( icount == icm .or. k==nlevs) then
+                call general_reload2(grd,g_z,g_ps,g_tv,g_vor,g_div,g_u,g_v,g_q,g_oz, &
+                     g_ql,g_qi,g_qr,g_qs,g_qg,icount,iflag,ilev,work,uvflag,vordivflag,g_ni,g_nr)
+                endif
+          enddo ! do k=1,nlevs
+       endif ! imp_physics
+!       do k=1,nlevs
```

Comment:
This section of code is commented out in `develop`.  This PR retains the commented out section of code?  Can we remove the commented out code on [lines 3655 to 3683](https://github.com/emilyhcliu/GSI/blob/3404ddfef03fd52c1fa04411d6613e15b7ef77da/src/gsi/general_read_gfsatm.f90#L3655-#L3683)?

If not, why do we need to retain it?

Reply:
> This section of code is commented out in `develop`. This PR retains the commented out section of code? Can we remove the commented out code on [lines 3655 to 3683](https://github.com/emilyhcliu/GSI/blob/3404ddfef03fd52c1fa04411d6613e15b7ef77da/src/gsi/general_read_gfsatm.f90#L3655-#L3683)?
> 
> If not, why do we need to retain it?

This code section is to read in cloud fraction, which is not a standard variable in the history file.
We could add the cloud amount (fraction) variable in the diagnostic variable table when we configure the parallel experiment so the cloud amount will be written out.  

I kept it there in case we need to test the cloud fraction from the model directly.  
The cloud fraction in GFS is not a prognostic variable but a diagnostic one. It is diagnosed as a function of grid box averaged temperature, moisture, and hydrometer mixing ratios. So, we have implemented this in GSI (outer loop).

Reply:
Subroutine `general_read_gfsatm_nems` sets logical flag `has_cf` to decide whether or not to execute the `cld_amt` block.  If we could do the same with the netcdf reader, we could uncomment the `cld_amt` block and `gsi.x` would be ready to read `cld_amt` whenever the field is available in the netcdf input file. However, it seems ncio does not offer an easy way to query the contents of a netcdf file.  This makes it more challenging to implement the nemsio approach.  

We can retain the commented out code for possible future use.